aten/src/ATen/native/cuda/SoftMax.cu

```diff
@@ -1005,8 +1008,71 @@ Tensor masked_softmax_cuda(const Tensor& input, const Tensor& mask) {
           mask.data_ptr<bool>()
         );
       });
-    }
-    return output;
+  }
+  return output;
+}
+
+Tensor masked_softmax_backward_cuda(
```

Comment:
Do you need to check if the mask_ is transformer layout in this case?

Reply:
As discussed offline, will resolve in a follow-up diff!